src/swap.rs

```diff
@@ -415,397 +31,112 @@ where
     let hbit_funded = alice.fund(&hbit_params).await?;
 
     if !bob.is_safe_to_fund(herc20_params.expiry).await? {
```

Comment:
I thought you were ok to rename those functions?

Reply:
I have gone for a slightly different approach on a separate branch which builds on top of this. These traits will be gone! ☺️